sdk/communication/AzureCommunicationCommon/Source/Identifiers.swift

```diff
@@ -75,17 +80,36 @@ import Foundation
      */
     public init(phoneNumber: String, rawId: String? = nil) {
         self.phoneNumber = phoneNumber
-        self.rawId = rawId
+        if let rawId = rawId {
+            self.rawId = rawId
+        } else {
+            let strippedPhoneNumber = phoneNumber.replacingOccurrences(of: "+", with: "")
+            self.rawId = "4:" + strippedPhoneNumber
+        }
     }
 
+    /**
+     Returns a Boolean value indicating whether two values are equal.
+        Note: In Objective-C favor isEqual() method
+     - Parameter lhs PhoneNumberIdentifier to compare.
+     - Parameter rhs  Another PhoneNumberIdentifier to compare.
+     */
+    // swiftlint:disable nsobject_prefer_isequal
     public static func == (lhs: PhoneNumberIdentifier, rhs: PhoneNumberIdentifier) -> Bool {
-        if lhs.phoneNumber != rhs.phoneNumber {
+        return lhs.rawId == rhs.rawId
```

Comment:
If we make this change we will be changing some existing behaviour. 
Are we ok with this? 
Is this considered a breaking change?

Reply:
are you talking about the case that someone initializes the `PhoneNumberIdentifier` with a `phoneNumber` and a non-corresponding `rawId`?

Reply:
not sure if we want to support that case at all. perhaps we could add a validation to the constructor to avoid this case.

Reply:
`are you talking about the case that someone initializes the PhoneNumberIdentifier with a phoneNumber and a non-corresponding rawId?`
Yes. We previously had unit tests to validate this one case. So as per this new PR we would remove this case. 
`not sure if we want to support that case at all. perhaps we could add a validation to the constructor to avoid this case.`
Yes this PR is make this change so it will no longer be possible

Reply:
@JoshuaLai I think the archboard would definitely consider changing the behavior a breaking change... _unless_ the original behavior was considered a bug :)

Reply:
We had supported to override the `phoneNumber` with a `rawId` previously and if rawId wsa provided the phoneNumber doesn't matter.
> unless the original behavior was considered a bug :)
So imo this is the case, and the change now is a bug fix.

> allowing the rawId to not correspond with the phone number almost seems like a bug to me.
It is a feature by design unfortunately because we were made aware of by our service teams that there are phone numbers that have a different format (no 4:).